src/sb_gazebo/robots/rover_2018.xacro

```diff
@@ -0,0 +1,287 @@
+    <xacro:property name="tower_upright_x_size" value="0.02"/>
+    <xacro:property name="tower_upright_y_size" value="0.02"/>
+    <xacro:property name="tower_upright_z_offset" value="${tower_upright_height/2 + chassis_z_size/2}"/>
+
+    <xacro:property name="camera_size" value="0.05"/>
+    <xacro:property name="camera_pitch" value="${pi/4}"/>
+    <xacro:property name="camera_x_offset" value="0.03"/>
+    <xacro:property name="camera_z_offset" value="${tower_upright_height/2}"/>
+
+    <!-- JOINTS AND LINKS
+         This section uses all the values calculated above to actually create
+         all the links and joints of the robot
+    -->
+
+    <!-- Footprint of the robot (because gazebo's base_link can't have interia) -->
+    <link name='base_link'/>
+    <joint name="base_joint" type="fixed">
+        <parent link="base_link"/>
+        <child link="chassis"/>
+        <origin xyz="
+            ${chassis_x_pos}
+            ${chassis_y_pos}
+            ${chassis_z_pos}
+            " 
+            rpy="0 0 0"
+            />
+    </joint>
+    <box_link 
+        name="chassis"
+        x_pos="0"
+        y_pos="0"
+        z_pos="0"
+        x_size="${chassis_x_size}"
+        y_size="${chassis_y_size}"
+        z_size="${chassis_z_size}"
+        roll="0" 
+        pitch="0" 
+        yaw="0"
+        mass="10"
+        material_name="grey"
+        />
+
+    <xacro:macro name="primary_rocker" params="lr_prefix y_reflect">
+        <!-- TODO: Make this revolute model with a spring to resist rotation (but allow it) instead of just being fixed like it currently is -->
+        <joint name="${lr_prefix}_primary_rocker_to_chassis" type="revolute">
+            <parent link="chassis"/>
+            <child link="${lr_prefix}_primary_rocker"/>
+            <origin xyz="
+                ${primary_rocker_x_offset} 
+                ${primary_rocker_y_offset*y_reflect} 
+                ${primary_rocker_z_offset}
+                " 
+                rpy="
+                0
+                0
+                0
+                " 
+                />
+            <axis xyz="0 1 0" rpy="0 0 0" />
+            <limit effort="100" velocity="100"/>
+            <joint_properties damping="0.0" friction="0.0"/>
+            <limit lower="-20" upper="20" effort="100" velocity="100"/>
+        </joint>
+        <cylinder_link 
+            name="${lr_prefix}_primary_rocker"
+            x_pos="0" 
+            y_pos="0" 
+            z_pos="0" 
+            radius="${primary_rocker_radius}"
+            length="${primary_rocker_length}"
+            roll="${pi/2}" 
+            pitch="${pi/2}" 
+            yaw="0"
+            mass="1"
+            material_name="grey"
+            />
+    </xacro:macro>
+    <primary_rocker lr_prefix="L" y_reflect="1"/>
+    <primary_rocker lr_prefix="R" y_reflect="-1"/>
+
+    <xacro:macro name="primary_rocker_arm" params="lr_prefix fb_prefix x_reflect y_reflect">
```

Comment:
I think it would be helpful to have high-level comments describing what is happening in each xacro block. For example in this block, the arms are being connected to the rocker (I think).

Reply:
It might also be cleaner to put each of these blocks into their own xacro files, and include them back with xacro:include. That way we would have xacro file for each part, so we could easily change a specific part without having to search through this long file.

Reply:
Added javadoc style comments to each macro.